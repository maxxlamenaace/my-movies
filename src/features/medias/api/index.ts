import { getMediaCredits } from './get-media-credits/get-media-credits';
import { getMediaDetails } from './get-media-details/get-media-details';
import { getMediaGenres } from './get-media-genres/get-media-genres';
import { getMediaImages } from './get-media-images/get-media-images';
import { getMediaRecommendations } from './get-media-recommendations/get-media-recommendations';
import { getMediaVideos } from './get-media-videos/get-media-videos';
import { getMediasList } from './get-medias-list/get-medias-list';

export {
  getMediaCredits,
  getMediaDetails,
  getMediaGenres,
  getMediaImages,
  getMediaRecommendations,
  getMediaVideos,
  getMediasList,
};
