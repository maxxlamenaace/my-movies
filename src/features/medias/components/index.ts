import CircularRate from './circular-rate/CircularRate';
import MediasAutoSwiper from './medias-auto-swiper/MediasAutoSwiper';
import MediasList from './medias-list/MediasList';
import MediaItem from './media-item/MediaItem';
import MediaBackgroundHeader from './media-background-header/MediaBackgroundHeader';
import MediaInfos from './media-infos/MediaInfos';
import CastSlide from './cast-slide/CastSlide';
import MediaVideosSlide from './media-videos-slide/MediaVideosSlide';
import MediaVideoItem from './media-video-item/MediaVideoItem';
import MediaImagesSlide from './media-images-slide/MediaImagesSlide';
import MediasGrid from './medias-grid/MediasGrid';
import FavoriteItem from './favorite-item/FavoriteItem';
import FavoritesGrid from './favorites-grid/FavoritesGrid';

export {
  CircularRate,
  MediasAutoSwiper,
  MediasList,
  MediaItem,
  MediaBackgroundHeader,
  MediaInfos,
  CastSlide,
  MediaVideosSlide,
  MediaVideoItem,
  MediaImagesSlide,
  MediasGrid,
  FavoriteItem,
  FavoritesGrid,
};
