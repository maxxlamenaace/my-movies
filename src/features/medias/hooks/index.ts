import { useMediaDetails } from './use-media-details/use-media-details';
import { useMediaGenres } from './use-media-genres/use-media-genres';
import { useAllMedias } from './use-all-medias/use-all-medias';
import { useMediasList } from './use-medias-list/use-medias-list';
import { useFavoriteMedias } from './use-favorite-medias/use-favorite-medias';

export { useMediaDetails, useAllMedias, useMediaGenres, useMediasList, useFavoriteMedias };
