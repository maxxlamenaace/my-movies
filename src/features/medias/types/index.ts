export type MediaType = 'movie' | 'tv';
export type MediaCategory = 'popular' | 'top_rated';
export type MediaImageType = 'poster' | 'backdrop';

export type Media = {
  id: number;
  poster_path?: string;
  adult: boolean;
  overview: string;
  genre_ids: number[];
  original_title: string;
  original_language: string;
  media_type?: MediaType;
  title?: string; // Only for movies
  name?: string; // Only for series
  backdrop_path?: string;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
  release_date?: string; // Only for movies
  first_air_date?: string; // Only for series
};

export type MediaDetails = {
  id: number;
  budget: number;
  genres: Genre[];
  homepage: string;
  revenue: number;
  runtime: number;
  status: string;
  tagline: string;
} & Media;

export type MediaCredit = {
  id: number;
  adult: boolean;
  gender: number;
  known_for_department: string;
  name: string;
  original_name: string;
  popularity: number;
  profile_path: string;
  cast_id: number;
  character: string;
  credit_id: string;
  order: number;
};

export type MediaVideo = {
  id: number;
  type: string;
  official: boolean;
  name: string;
  site: string;
  size: number;
  published_at: Date;
  key: string;
};

export type MediaImage = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

/* export type ProductionCompany = {
  id: number;
  logo_path: string;
  name: string;
  origin_country: string;
};

export type ProductionCountry = {
  iso_3166_1: string;
  name: string;
};

export type SpokenLanguage = {
  iso_639_1: string;
  name: string;
}; */

export type Genre = {
  id: number;
  name: string;
};
