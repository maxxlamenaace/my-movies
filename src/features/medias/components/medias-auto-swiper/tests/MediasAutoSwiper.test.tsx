import React from 'react';
import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { media, genre } from '@/tests/factories';
import { MediasAutoSwiper } from '@/features/medias';
import { act } from 'react-dom/test-utils';

describe('MediasAutoSwiper', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', async () => {
    const { asFragment } = render(
      <MediasAutoSwiper medias={[media()]} mediaType='movie' genres={[genre()]} />,
      { wrapper: BrowserRouter },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('when user clicks on watch now button', () => {
    it('redirects to the media details page', () => {
      const mediaToBeDisplayed = media();
      const { getAllByText } = render(
        <MediasAutoSwiper medias={[mediaToBeDisplayed]} mediaType='movie' genres={[genre()]} />,
        { wrapper: BrowserRouter },
      );

      act(() => {
        fireEvent.click(getAllByText('Watch now')[0]);
      });

      waitFor(() => {
        expect(global.window.location.pathname).toContain(`medias/movie/${mediaToBeDisplayed.id}`);
      });
    });
  });
});
