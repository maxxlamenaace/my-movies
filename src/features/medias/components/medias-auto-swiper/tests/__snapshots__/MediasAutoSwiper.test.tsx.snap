// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediasAutoSwiper renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-124i1uv"
  >
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events media-auto-swiper"
      role="swiper"
      style="width: 100%;"
    >
      <div
        class="swiper-wrapper"
        style="cursor: grab; transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="0"
        >
          <div
            class="MuiBox-root css-1td14js"
          />
          <div
            class="MuiBox-root css-s9z9b"
          />
          <div
            class="MuiBox-root css-1e2m3rn"
          >
            <div
              class="MuiBox-root css-9jhkvv"
            >
              <div
                class="css-678bp8-MuiStack-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-mrmr51-MuiTypography-root"
                >
                  title
                </h4>
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <div
                    class="MuiBox-root css-d2o2iw"
                  >
                    <span
                      aria-valuenow="90"
                      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSuccess css-zc2y38-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 50px; height: 50px; transform: rotate(-90deg);"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                          style="stroke-dasharray: 126.920; stroke-dashoffset: 12.692px;"
                        />
                      </svg>
                    </span>
                    <div
                      class="MuiBox-root css-15i2x3e"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-caption css-1dtdxxd-MuiTypography-root"
                      >
                        9
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
                  />
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    />
                  </div>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-14ndvjh-MuiTypography-root"
                >
                  overview
                </p>
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1t6l1gc-MuiButtonBase-root-MuiButton-root"
                  href="/medias/movie/2005013077"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PlayArrowIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 5v14l11-7z"
                      />
                    </svg>
                  </span>
                  Watch now
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="swiper-slide"
          data-swiper-slide-index="0"
        >
          <div
            class="MuiBox-root css-1td14js"
          />
          <div
            class="MuiBox-root css-s9z9b"
          />
          <div
            class="MuiBox-root css-1e2m3rn"
          >
            <div
              class="MuiBox-root css-9jhkvv"
            >
              <div
                class="css-678bp8-MuiStack-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-mrmr51-MuiTypography-root"
                >
                  title
                </h4>
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <div
                    class="MuiBox-root css-d2o2iw"
                  >
                    <span
                      aria-valuenow="90"
                      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSuccess css-zc2y38-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 50px; height: 50px; transform: rotate(-90deg);"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                          style="stroke-dasharray: 126.920; stroke-dashoffset: 12.692px;"
                        />
                      </svg>
                    </span>
                    <div
                      class="MuiBox-root css-15i2x3e"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-caption css-1dtdxxd-MuiTypography-root"
                      >
                        9
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
                  />
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    />
                  </div>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-14ndvjh-MuiTypography-root"
                >
                  overview
                </p>
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1t6l1gc-MuiButtonBase-root-MuiButton-root"
                  href="/medias/movie/2005013077"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PlayArrowIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 5v14l11-7z"
                      />
                    </svg>
                  </span>
                  Watch now
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="0"
        >
          <div
            class="MuiBox-root css-1td14js"
          />
          <div
            class="MuiBox-root css-s9z9b"
          />
          <div
            class="MuiBox-root css-1e2m3rn"
          >
            <div
              class="MuiBox-root css-9jhkvv"
            >
              <div
                class="css-678bp8-MuiStack-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-mrmr51-MuiTypography-root"
                >
                  title
                </h4>
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <div
                    class="MuiBox-root css-d2o2iw"
                  >
                    <span
                      aria-valuenow="90"
                      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSuccess css-zc2y38-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 50px; height: 50px; transform: rotate(-90deg);"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                          style="stroke-dasharray: 126.920; stroke-dashoffset: 12.692px;"
                        />
                      </svg>
                    </span>
                    <div
                      class="MuiBox-root css-15i2x3e"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-caption css-1dtdxxd-MuiTypography-root"
                      >
                        9
                      </div>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
                  />
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    />
                  </div>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-14ndvjh-MuiTypography-root"
                >
                  overview
                </p>
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1t6l1gc-MuiButtonBase-root-MuiButton-root"
                  href="/medias/movie/2005013077"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PlayArrowIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 5v14l11-7z"
                      />
                    </svg>
                  </span>
                  Watch now
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
